propagation.go

```diff
@@ -65,17 +65,17 @@ const (
 	//    carrier := HTTPHeaderTextMapCarrier(httpReq.Header)
 	//    err := span.Tracer().Inject(span, TextMap, carrier)
 	//
-	// Or Join():
+	// Or Extract():
 	//
 	//    carrier := HTTPHeaderTextMapCarrier(httpReq.Header)
-	//    span, err := tracer.Join("opName", TextMap, carrier)
+	//    span, err := tracer.Extract("opName", TextMap, carrier)
 	//
 	TextMap
 )
 
 // TextMapWriter is the Inject() carrier for the TextMap builtin format. With
-// it, the caller can encode a Span for propagation as entries in a multimap of
-// unicode strings.
+// it, the caller can encode a SpanMetadata for propagation as entries in a
+// multimap of unicode strings.
```

Comment:
I missed this before - I think we must not require multi-map semantics from the TextMap carrier implementation, and tracers should not expect the underlying transport to support multi-map headers.